pilot/pkg/networking/core/v1alpha3/networkfilter_test.go

```diff
@@ -105,6 +107,37 @@ func TestInboundNetworkFilterStatPrefix(t *testing.T) {
 	}
 }
 
+// Test that the metadata exchange filter comes before RBAC network filter
+func TestInboundNetworkFilterOrder(t *testing.T) {
+	services := []*model.Service{
+		buildService("test.com", "10.10.0.0/24", protocol.TCP, tnow),
+	}
+	t.Run("mx-filter-before-rbac-filter", func(t *testing.T) {
+		cg := NewConfigGenTest(t, TestOptions{
+			Services: services,
+		})
+
+		fcc := inboundChainConfig{
+			clusterName: "inbound|8888||",
+		}
+		push := cg.PushContext()
+		push.AuthzPolicies = getAuthorizationPolicies()
+		proxy := node(nil)
+		listenerFilters := NewListenerBuilder(proxy, push).buildInboundNetworkFilters(fcc)
+
+		rbacFilterFound := false
+		RBACTCPFilterName := "envoy.filters.network.rbac"
+
+		for _, filter := range listenerFilters {
```

Comment:
nit: rather than this bespoke check maybe it would make sense to just assert what filter should be present?

```
					listenertest.VerifyFilterChain(t, fc, listenertest.FilterChainTest{
						NetworkFilters: []string{wellknown.TCPProxy},
						TotalMatch:     true,
					})
```

is an example from existing code doing this

Reply:
Makes sense.